src/rdkafka_buf.h

```diff
@@ -399,6 +399,11 @@ struct rd_kafka_buf_s { /* rd_kafka_buf_t */
                 } Metadata;
                 struct {
                         rd_kafka_msgbatch_t batch; /**< MessageSet/batch */
+                        rd_list_t batch_list;      /* MessageSet/batch list*/
```

Comment:
This should be a map no ? https://github.com/a0x8o/kafka/blob/master/clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java#L1217C9-L1217C114

Reply:
It totally should. My bad, I picked the list for dev speed, since I have to use it already, and I didn't have time to read though rdmap.c to understand its implementation and memory management *sweat smile*.
It's inefficient on the Response processing path when you need to look up by topic + partition-id from the response, but we won't have 100s of batches in a response. So it's okey for a first try